WinUIGallery/Samples/ControlPages/CaptureElementPreviewPage.xaml.cs

```diff
@@ -55,30 +60,68 @@ private void CaptureElementPreviewPage_Unloaded(object sender, RoutedEventArgs e
     private MediaFrameSourceGroup mediaFrameSourceGroup;
     private MediaCapture mediaCapture;
 
-    async private void StartCaptureElement()
+    private async Task StartCaptureElement()
     {
-        var groups = await MediaFrameSourceGroup.FindAllAsync();
-        if (groups.Count == 0)
+        try
         {
-            frameSourceName.Text = "No camera devices found.";
-            return;
+            var groups = await MediaFrameSourceGroup.FindAllAsync();
+            if (groups.Count == 0)
+            {
+                frameSourceName.Text = "No camera devices found.";
+                return;
+            }
+            mediaFrameSourceGroup = groups.First();
+
+            frameSourceName.Text = "Viewing: " + mediaFrameSourceGroup.DisplayName;
+            mediaCapture = new MediaCapture();
+            var mediaCaptureInitializationSettings = new MediaCaptureInitializationSettings()
+            {
+                SourceGroup = this.mediaFrameSourceGroup,
+                SharingMode = MediaCaptureSharingMode.SharedReadOnly,
+                StreamingCaptureMode = StreamingCaptureMode.Video,
+                MemoryPreference = MediaCaptureMemoryPreference.Cpu
+            };
+            await mediaCapture.InitializeAsync(mediaCaptureInitializationSettings);
+
+            // Set the MediaPlayerElement's Source property to the MediaSource for the mediaCapture.
+            var frameSource = mediaCapture.FrameSources[this.mediaFrameSourceGroup.SourceInfos[0].Id];
+            captureElement.Source = Windows.Media.Core.MediaSource.CreateFromMediaFrameSource(frameSource);
         }
-        mediaFrameSourceGroup = groups.First();
-
-        frameSourceName.Text = "Viewing: " + mediaFrameSourceGroup.DisplayName;
-        mediaCapture = new MediaCapture();
-        var mediaCaptureInitializationSettings = new MediaCaptureInitializationSettings()
+        catch (UnauthorizedAccessException)
+        {
+            var hyperlink = new Hyperlink() { Inlines = { new Run() { Text = "camera settings" } } };
+            hyperlink.Click += async (_, _) => await Launcher.LaunchUriAsync(new Uri("ms-settings:privacy-webcam"));
+
+            var dialog = new ContentDialog()
+            {
+                Title = "Camera access denied",
+                Content = new TextBlock()
+                {
+                    Inlines =
+                    {
+                        new Run() { Text = "Please check your " },
```

Comment:
Nit: would it make sense to make the primary button? and the secondary button is to cancel the dialog?

Reply:
Yeah, it's better with the primary button. Please take a look.